arelle/CntlrCmdLine.py

```diff
@@ -744,17 +753,33 @@ def monitorParentProcess():
             self.setLogLevelFilter(options.logLevelFilter)
         if options.logCodeFilter:
             self.setLogCodeFilter(options.logCodeFilter)
+        from arelle.ValidateXbrlCalcs import ValidateCalcsMode as CalcsMode
         if options.calcDecimals:
             if options.calcPrecision:
                 self.addToLog(_("both --calcDecimals and --calcPrecision validation are requested, proceeding with --calcDecimals only"),
                               messageCode="info", file=options.entrypointFile)
-            self.modelManager.validateInferDecimals = True
-            self.modelManager.validateCalcLB = True
+            if options.calcs:
+                self.addToLog(_("both --calcDecimals and --calcs validation are requested, proceeding with --calcDecimals only"),
+                              messageCode="info", file=options.entrypointFile)
+            self.modelManager.validateCalcs = CalcsMode.XBRL_v2_1 + (options.calcDeduplicate or 0)
         elif options.calcPrecision:
-            self.modelManager.validateInferDecimals = False
-            self.modelManager.validateCalcLB = True
-        if options.calcDeduplicate:
-            self.modelManager.validateDedupCalcs = True
+            self.modelManager.validateCalcs = CalcsMode.XBRL_v2_1_INFER_PRECISION
+        else:
+            try:
+                self.modelManager.validateCalcs = {
+                     "xbrl21precision": CalcsMode.XBRL_v2_1_INFER_PRECISION,
+                     "xbrl21": CalcsMode.XBRL_v2_1,
+                     "c10": CalcsMode.XBRL_v2_1,
+                     "xbrl21-dedup": CalcsMode.XBRL_v2_1_DEDUPLICATE,
+                     "c10d": CalcsMode.XBRL_v2_1_DEDUPLICATE,
+                     "round-to-nearest": CalcsMode.ROUND_TO_NEAREST,
+                     "c11r": CalcsMode.ROUND_TO_NEAREST,
+                     "truncation": CalcsMode.TRUNCATION,
+                     "c11t": CalcsMode.TRUNCATION
+                    }.get(options.calcs)
+            except KEYERROR:
+                self.addToLog(_("--calcs value invalid, request ignored"),
+                              messageCode="info", file=options.entrypointFile)
```

Comment:
get won't throw a key error here:
```suggestion
                    }[options.calcs]
            except KEYERROR:
                self.addToLog(_("--calcs value invalid, request ignored"),
                              messageCode="info", file=options.entrypointFile)
```